crates/driver/src/build_graph.rs

```diff
@@ -506,11 +523,16 @@ impl GraphState {
                 if no_loops.contains(node) {
                     continue;
                 }
-                if self.in_cycle(*node) {
-                    incompress = Some(*node);
-                    break 'outer_loop;
-                } else {
-                    no_loops.insert(*node);
+                match self.in_cycle(*node, circular_dependency_allow_list) {
```

Comment:
I think you can do:

```rust
let cycle = self.in_cycle(*node, circular_dependency_allow_list)?;
if cycle {

}
else ...
```
or:
```rust
match self.in_cycle(*node, circular_dependency_allow_list) {
  Ok(true) =>  {
                            incompress = Some(*node);
                            break 'outer_loop;
  }
  Ok(false) => no_loops.insert(*node);
  Err(e) => return Err(e)
}
```
I tend to think the `?` one is more idiomatic.

Reply:
actually, I think `if self.in_cycle(*node, circular_dependency_allow_list)? {` will work if you want to make a more minimal change, and it might just be a one liner at this point.

Reply:
Yea, `?` totally works fine - https://github.com/bazeltools/bzl-gen-build/pull/384/commits/a169ea0a26e328bd78843b078e6f0bc33cfc845a